spark/src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -234,9 +234,14 @@ class DeltaTableHadoopOptionsSuite extends QueryTest
   private val publicMethods =
   scala.reflect.runtime.universe.typeTag[io.delta.tables.DeltaTable].tpe.decls
     .filter(_.isPublic)
+    .filter(!_.isSynthetic)
     .map(_.name.toString).toSet
 
-  private val ignoreMethods = Seq()
+  private val ignoreMethods = Seq(
+    "clone",
+    "cloneAtTimestamp",
+    "cloneAtVersion",
+  )
```

Comment:
I wasn't sure what to do regarding the Hadoop conf for cloning. It seems like the options are:
- Ignore the conf (which is what this is currently doing)
- Use the table conf for the source table, but use the default conf for the target table
- Use the table conf for the source table and the target table
- Use the table conf for the source table, and add a parameter to the clone functions for custom configs for the target

Reply:
@Kimahriman Hey Adam, what is the Hadoop Conf that you are referring to here and why would you need it?

Reply:
I think the whole purpose of `DeltaTableHadoopOptionsSuite` is to make sure all `io.delta.DeltaTable` functions respect the `hadoopConf` passed into functions like `DeltaTable.forPath`, but it's unclear exactly if/how that should be used for cloning purposes

Reply:
@Kimahriman I don't think you need to include the Hadoop conf in the API.

This has already been handled here:
1.  https://github.com/delta-io/delta/blob/51ecfe5ef1b759e88d7e799b94c1919c4aa9eef0/spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableCommand.scala#L99
2. https://github.com/delta-io/delta/blob/51ecfe5ef1b759e88d7e799b94c1919c4aa9eef0/spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala#L183